sql: regional_by_row_hash_sharded_index_query_plan logic test failed in CCL

Extracted from https://github.com/cockroachdb/cockroach/issues/75545#issuecomment-1050860894.

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4458473&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4458473&tab=artifacts#/) on master @ [95b0430c876abc9a204871c6391f0e4f2f2df910](https://github.com/cockroachdb/cockroach/commits/95b0430c876abc9a204871c6391f0e4f2f2df910):


```
=== RUN   TestStoreReadInconsistent/INCONSISTENT
    logic.go:2479: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_hash_sharded_index_query_plan:40: EXPLAIN (VERBOSE) INSERT INTO t_child VALUES (123, 321);
        expected:
            distribution: local
            vectorized: true
            ·
            • root
            │ columns: ()
            │
            ├── • insert
            │   │ columns: ()
            │   │ estimated row count: 0 (missing stats)
            │   │ into: t_child(id, pid)
            │   │
            │   └── • buffer
            │       │ columns: (column1, column2)
            │       │ label: buffer 1
            │       │
            │       └── • values
            │             columns: (column1, column2)
            │             size: 2 columns, 1 row
            │             row 0, expr 0: 123
            │             row 0, expr 1: 321
            │
            └── • constraint-check
                │
                └── • error if rows
                    │ columns: ()
                    │
                    └── • lookup join (anti)
                        │ columns: (column2)
                        │ estimated row count: 0 (missing stats)
                        │ table: t_parent@t_parent_pkey
                        │ equality cols are key
                        │ lookup condition: ((column2 = id) AND (crdb_region IN ('ca-central-1', 'us-east-1'))) AND (crdb_internal_id_shard_16 IN (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15))
                        │
                        └── • lookup join (anti)
                            │ columns: (column2)
                            │ estimated row count: 1 (missing stats)
                            │ table: t_parent@t_parent_pkey
                            │ equality cols are key
                            │ lookup condition: ((column2 = id) AND (crdb_region = 'ap-southeast-2')) AND (crdb_internal_id_shard_16 IN (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15))
                            │
                            └── • project
                                │ columns: (column2)
                                │ estimated row count: 1
                                │
                                └── • scan buffer
                                      columns: (column1, column2)
                                      label: buffer 1
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • root
            │ columns: ()
            │
            ├── • insert
            │   │ columns: ()
            │   │ estimated row count: 0 (missing stats)
            │   │ into: t_child(id, pid)
            │   │
            │   └── • buffer
            │       │ columns: (column1, column2)
            │       │ label: buffer 1
            │       │
            │       └── • values
            │             columns: (column1, column2)
            │             size: 2 columns, 1 row
            │             row 0, expr 0: 123
            │             row 0, expr 1: 321
            │
            └── • constraint-check
                │
                └── • error if rows
                    │ columns: ()
                    │
                    └── • lookup join (anti)
                        │ columns: (column2)
                        │ estimated row count: 0 (missing stats)
                        │ table: t_parent@t_parent_pkey
                        │ equality cols are key
                        │ lookup condition: ((column2 = id) AND (crdb_region IN ('ap-southeast-2', 'ca-central-1', 'us-east-1'))) AND (crdb_internal_id_shard_16 IN (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15))
                        │
                        └── • project
                            │ columns: (column2)
                            │ estimated row count: 1
                            │
                            └── • scan buffer
                                  columns: (column1, column2)
                                  label: buffer 1
    logic.go:2022: 
         /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_hash_sharded_index_query_plan:91: too many errors encountered, skipping the rest of the input
[12:49:40] --- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_hash_sharded_index_query_plan with config multiregion-9node-3region-3azs: 7 tests, 2 failures

[12:49:42] --- total progress: 2116 statements
            --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_hash_sharded_index_query_plan/test_fk_constraint (0.49s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_hash_sharded_index_query_plan
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_hash_sharded_index_query_plan (5.34s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (380.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>



Jira issue: CRDB-13385

--- a/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_hash_sharded_index_query_plan
+++ b/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_hash_sharded_index_query_plan
@@ -36,7 +36,7 @@ CREATE TABLE t_child_regional (
 # TODO (mgartner): there is a lookup join with lookup condition checking every
 # single shard. This is unnecessary and could be improved by having the shard
 # number calculated instead of looking at all possible shards.
-query T
+query T retry
 EXPLAIN (VERBOSE) INSERT INTO t_child VALUES (123, 321);
 ----
 distribution: local

